File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/image/ImageUpdateDto.java
package com.intexsoft.webshop.productservice.dto.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.annotation.Nullable;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
public class ImageUpdateDto {
    @Nullable
    @Positive
    Long id;
    @Nullable
    @Size(max = 512)
    @Pattern(regexp = "^(?!\\s*$).+")
    String filePath;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/image/ImageUpdateDto.kt
package com.intexsoft.webshop.productservicekt.dto.image

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.Size
@JsonIgnoreProperties(ignoreUnknown = true)
data class ImageUpdateDto(
    @field:Positive
    val id: Long?,
    @field:NotBlank
    @field:Size(max = 512)
    val filePath: String?
)